components/user_settings/email_notification_setting.jsx

```diff
@@ -27,35 +27,49 @@ export default class EmailNotificationSetting extends React.Component {
 
     constructor(props) {
         super(props);
+        this.enableEmail = props.enableEmail;
+        this.emailInterval = props.emailInterval;
 
         this.state = {
             enableEmail: props.enableEmail,
-            emailInterval: props.emailInterval
+            emailInterval: props.emailInterval,
+            hasChanged: false
         };
     }
 
     componentWillReceiveProps(nextProps) {
         if (nextProps.enableEmail !== this.props.enableEmail || nextProps.emailInterval !== this.props.emailInterval) {
+            this.enableEmail = nextProps.enableEmail;
+            this.emailInterval = nextProps.emailInterval;
             this.setState({
                 enableEmail: nextProps.enableEmail,
-                emailInterval: nextProps.emailInterval
+                emailInterval: nextProps.emailInterval,
+                hasChanged: false
             });
         }
     }
 
     handleChange = (enableEmail, emailInterval) => {
+        const hasChanged = this.enableEmail !== enableEmail || this.emailInterval !== emailInterval;
```

Comment:
Do we need class variables for this? Why not just do something like:
```
this.props.enableEmail !== enableEmail || this.props.emailInterval !== emailInterval
```

Also `hasChanged` isn't used for rendering so I don't think it really needs to be part of state, why not just do the compare inside `handleSubmit()`?

Reply:
I initially thought it would be good for minor convenience only, but I agree, it's not necessary. Please see updated PR. thanks!